src/mod/irc.mod/msgcmds.c

```diff
@@ -431,7 +431,7 @@ static int msg_who(char *nick, char *host, struct userrec *u, char *par)
 
 static int msg_whois(char *nick, char *host, struct userrec *u, char *par)
 {
-  char s[UHOSTLEN], s1[81], *s2;
+  char s[UHOSTLEN + 1], s1[143], *s2, stime[14];
```

Comment:
s[NICKLEN + UHOSTLEN + 1]?

Reply:
Similar to my comment above.
Here are the compiler warnings that would occur, if snprintf instead of egg_snprintf would have been used with a destination buffer of only UHOSTLEN:
```
gcc -fPIC -g -O2 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/usr/include -g3 -DDEBUG -DDEBUG_ASSERT -DDEBUG_MEM -DDEBUG_DNS  -DMAKING_MODS -c .././irc.mod/irc.c && mv -f irc.o ../
In file included from .././irc.mod/irc.c:69:
.././irc.mod/msgcmds.c: In function ‘msg_whois’:
.././irc.mod/msgcmds.c:465:37: warning: ‘snprintf’ output may be truncated before the last format character [-Wformat-truncation=]
  465 |         snprintf(s, sizeof s, "%s!%s", par, m->userhost);
      |                                     ^
.././irc.mod/msgcmds.c:465:9: note: ‘snprintf’ output 2 or more bytes (assuming 325) into a destination of size 324
  465 |         snprintf(s, sizeof s, "%s!%s", par, m->userhost);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.././irc.mod/msgcmds.c: In function ‘msg_who’:
.././irc.mod/msgcmds.c:385:31: warning: ‘%s’ directive output may be truncated writing up to 323 bytes into a region of size between 291 and 323 [-Wformat-truncation=]
  385 |     snprintf(s, sizeof s, "%s!%s", m->nick, m->userhost);
      |                               ^~
.././irc.mod/msgcmds.c:385:5: note: ‘snprintf’ output between 2 and 357 bytes into a destination of size 324
  385 |     snprintf(s, sizeof s, "%s!%s", m->nick, m->userhost);
      |
```